com.playeveryware.eos/Runtime/Core/Utility/FileSystemUtility.cs

```diff
@@ -478,15 +499,235 @@ public static string ReadAllText(string path)
+        /// <returns>The filename component of the given path.</returns>
+        public static string GetFileName(string path)
+        {
+            return Path.GetFileName(path);
+        }
+
+        #endregion
+
+        #region Write Functionality
+
+        /// <summary>
+        /// Synchronously writes the given file contents to the given filepath,
+        /// optionally creating the directory structure if it does not already
+        /// exist.
+        /// </summary>
+        /// <param name="filePath">The path to write to.</param>
+        /// <param name="content">The contents to write to the file.</param>
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        public static void WriteFile(string filePath, string content, bool createDirectory = true)
+        {
+            // Use discard in-order to call async function without awaiting.
+            _ = WriteFileInternal(filePath, content, createDirectory);
+        }
+
+        /// <summary>
+        /// Asynchronously writes the given file contents to the given filepath,
+        /// optionally creating the directory structure if it does not already
+        /// exist.
+        /// </summary>
+        /// <param name="filePath">The path to write to.</param>
+        /// <param name="content">The contents to write to the file.</param>
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        /// <returns>Task</returns>
+        public static async Task WriteFileAsync(string filePath, string content, bool createDirectory = true)
+        {
+            await WriteFileInternal(filePath, content, createDirectory);
+        }
+
+        /// <summary>
+        /// Asynchronously writes the given file contents to the given filepath,
+        /// optionally creating the directory structure if it does not already
+        /// exist.
+        /// </summary>
+        /// <param name="filePath">The path to write to.</param>
+        /// <param name="content">The contents to write to the file.</param>
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        /// <returns>Task</returns>
+        private static async Task WriteFileInternal(string filePath, string content, bool createDirectory = true)
+        {
+            FileInfo file = new(filePath);
+
+            // If the directory should be created, and the DirectoryInfo is not
+            // null, then create the directory.
+            if (createDirectory && null != file.Directory)
+            {
+                CreateDirectory(file.Directory);
+            }
+
+            await using StreamWriter writer = new(filePath);
```

Comment:
Interesting, I've never considered awaiting a StreamWriter. Learned something from this.